src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs

```diff
@@ -65,7 +64,8 @@ internal abstract class PENamedTypeSymbol : NamedTypeSymbol
 
         private NamedTypeSymbol _lazyBaseType = ErrorTypeSymbol.UnknownResultType;
         private ImmutableArray<NamedTypeSymbol> _lazyInterfaces = default(ImmutableArray<NamedTypeSymbol>);
-        private NamedTypeSymbol _lazyDeclaredBaseType = ErrorTypeSymbol.UnknownResultType;
+        private NamedTypeSymbol _lazyDeclaredBaseTypeWithoutNullability = ErrorTypeSymbol.UnknownResultType;
```

Comment:
>private NamedTypeSymbol _lazyDeclaredBaseTypeWithoutNullability = ErrorTypeSymbol.UnknownResultType; [](start = 8, length = 100)

It doesn't look like adding this cache is necessary. Inside ```TypeKind.get``` we could check if we already have ```_lazyDeclaredBaseType``` and use that, otherwise we could resolve the type from metadata without applying any transforms, including tuple and dynamic, and use that type.

Reply:
Even better might be to bundle resolution of base type with calculating type kind for non-interfaces. Before we apply any transforms to the base type, calculate and set the type kind, then proceed with transforms. 

---
In reply to: [294935242](https://github.com/dotnet/roslyn/pull/36519#discussion_r294935242) [](ancestors = 294935242)